### DISCUSSION TITLE
Completing the checkout

### DISCUSSION THREAD

**Discussion by jannahalka** [saleor/saleor#11532] [Q&A] [answered]
Hi, I am new to saleor and I've been trying to use orderCreateFromCheckout mutation. I get this error when I try to trigger complete cart 

message | "You need one of the following permissions: HANDLE_CHECKOUTS"

This is my query: 
`mutation CheckoutComplete($checkoutId: ID!) {
    orderCreateFromCheckout(id: $checkoutId, removeCheckout: true) {
        order {
            id
            status
        }
        errors {
            field
            message
        }
    }
}`

I would like to ask If anyone can help me with this since I don't really know what am I supposed to do. Thanks

**Comment by korycins** [comment, marked as answer]
Hey, you're using the mutation dedicated to the Saleor App that handles the checkout flow - https://docs.saleor.io/docs/3.x/developer/checkout#creating-order-from-checkout
Here you have a more details about using checkout with built-in payment plugins: https://docs.saleor.io/docs/3.x/developer/checkout#finalizing-checkout-with-saleors-payment-gateways
And here is a section of handling the finalisation of checkout by Saleor App: https://docs.saleor.io/docs/3.x/developer/checkout#finalizing-checkout-with-saleor-app

**Reply by jannahalka** [reply]
Hi, I intended to use payment method that would use cash on delivery method so I didn't want to use plugins for credit card. Is there a plugin that would handle this situation? Because I would like to create a scenario where I could complete checkout without having credit card method.

**Reply by timuric** [reply]
This flow allows you to create orders and write transactions without use of payment plugins
https://docs.saleor.io/docs/3.x/developer/checkout#finalizing-checkout-with-saleor-app
